src/docfx/restore/RestoreMap.cs

```diff
@@ -1,69 +1,82 @@
 // Copyright (c) Microsoft. All rights reserved.
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
+using System.Collections.Concurrent;
 using System.Diagnostics;
 using System.IO;
+using System.Linq;
 
 namespace Microsoft.Docs.Build
 {
     internal class RestoreMap
     {
-        private readonly RestoreLock _restoreLock;
+        private static readonly ConcurrentDictionary<string, string> s_mappings = new ConcurrentDictionary<string, string>();
```

Comment:
Should it be as below to make each value created once?
```csharp
ConcurrentDictionary<string, Lazy<string>>
```

Reply:
good point, will change

---
In reply to: [230617031](https://github.com/dotnet/docfx/pull/3570#discussion_r230617031) [](ancestors = 230617031)